packages/components/src/menu/checkbox-item.tsx

```diff
@@ -26,16 +26,22 @@ export const MenuCheckboxItem = forwardRef<
 ) {
 	const menuContext = useContext( MenuContext );
 
+	if ( ! menuContext?.store ) {
+		throw new Error(
+			'Menu.CheckboxItem can only be rendered inside a Menu component'
+		);
```

Comment:
Just for context to anyone else reading this: throwing an error in this scenario makes sense even if we just look at Ariakit - it is consistent with how Ariakit does it:

https://github.com/ariakit/ariakit/blob/22b84f02b85bbc8c66b2c6b58cf947f24a5316ce/packages/ariakit-react-core/src/menu/menu-item.tsx#L77-L81

https://github.com/ariakit/ariakit/blob/22b84f02b85bbc8c66b2c6b58cf947f24a5316ce/packages/ariakit-core/src/utils/misc.ts#L217-L232

As you can see, if we don't have this check, Ariakit will throw this error:

```
MenuItem must be wrapped in a MenuList, Menu or Menubar component
```

however, relying on this would be confusing, since our `Menu` components don't necessarily follow the same naming scheme.